util/encounter_tools.py

```diff
@@ -142,7 +145,7 @@ def is_tl_line_begincast(poss_match):
     return re.search(r'^:([0-9A-F\(\)\|]+):([^:]+)', poss_match)
 
 def is_tl_line_buff(poss_match):
-    return re.search(r'1A:.......:(.+) gains the effect of (.+)( from)?', poss_match)
+    return re.search(r'1A:........:(.+) gains the effect of (.+)( from)?', poss_match)
```

Comment:
Will `(.+?)` fix this?

Reply:
I don't think so? I might have an error somewhere then. As it stands, it looks like the current regex [captures "[EFFECT] from [SOURCE] for [FLOAT] seconds"]( https://regex101.com/r/LPsWtX/2/). If I'm understanding correctly, [that change would capture only the first letter of [EFFECT]](https://regex101.com/r/0o1aC8/3/).

It would definitely be nice to extract that cleanly with a regex rather than having to run an additional string operation!

Reply:
It's because `from` is optional.  https://regex101.com/r/0o1aC8/4

Reply:
Hmm... that seems like it might work. I'll look a little more after TEA.

Reply:
I think maybe I misunderstood where we were trying to go with this. What you linked there would match only case 1:
1. `1A:........:Thancred gains the effect of Regen from Alphinaud for 15 seconds`
2. `1A:........:Thancred gains the effect of Regen from Alphinaud`
3. `1A:........:Thancred gains the effect of Regen from`
4. `1A:........:Thancred gains the effect of Regen`

Ideally, (I think) we want to be able to match cases 2 and 4 here. (Case 3 should never show up in a timeline, and it's going to be rare enough to see case 1 in a timeline that I think it's worth assuming that will just be done manually.) If possible, it would be nice to extract the effect cleanly from 2 as well, but if we can't both match and extract cleanly in a single regex it's probably not too big a deal.

(Looking at this more, it seems that trying to do what I'm getting at here with matching both 2 and 4 would involve some unpleasant manipulation of lookaheads, and I'm thinking that probably isn't worth the effort.)